js/filters.js

```diff
@@ -143,14 +143,22 @@ function TypeaheadFilter(elm) {
   Filter.call(this, elm);
 
   var key = this.$body.data('dataset');
+  var allowText = this.$body.data('allow-text') !== undefined;
   var dataset = typeahead.datasets[key];
-  this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
+  this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset, allowText);
```

Comment:
Or, should this be expressed differently, e.g.:  `this.$body.data('allow-text') !== undefined ? this.$body.data('allow-text') : false;`?